pkg/dictbuilder/dictbuilder.py

```diff
@@ -0,0 +1,54 @@
+import yaml
+from os import path
+import nltk
+
+
+class DictBuilder:
+    BASE_STOPWORDS = nltk.corpus.stopwords.words("english")
+
+    def __init__(
+        self, corpus_handle, tokenizer=nltk.TweetTokenizer(), enable_stopwords=False
+    ):
+        self.tokenizer = tokenizer
+        self.corpus_handle = corpus_handle
+        self.terms = set()
+        self.enable_stopwords = enable_stopwords
+        self.outfile_path = path.relpath(
+            path.join(
+                path.realpath(__file__),
+                "..",
+                "..",
+                "..",
+                "data",
+                "dictionary",
+                "UofOCourses.txt",
+            )
+        )
+
+    def build(self):
+        corpus_stream = yaml.load_all(self.corpus_handle, Loader=yaml.Loader)
+        for doc in corpus_stream:
+            self.terms = self.terms.union(self.tokenizer.tokenize(doc.course.contents))
+
+        out = open(self.outfile_path, "w")
+
+        rejected_terms = []
+        for term in sorted(map(lambda x: x.lower(), self.terms)):
+            if not self.enable_stopwords or not term in type(self).BASE_STOPWORDS:
+                out.write(term)
+                out.write("\n")
+            else:
+                rejected_terms.append(term)
+
+        if len(rejected_terms) > 0:
+            print(
+                "***The following terms were rejected from the dictionary since they have been identified as STOPWORDS***"
+            )
+            print(
+                "--------------------------------------------------------------------------------------------------------"
+            )
```

Comment:
If one of us is feeling up for it, we can probably make a logger module and just import that into every package. Though, tbh `print` statements are probably fine for now since we only use them for debugging (though it is nice to be able to log at the `Fatal` and `Warn` level sometimes"